MREUnityRuntime/MREUnityRuntimeLib/Core/Physics/JitterBuffer.cs

```diff
@@ -94,232 +103,196 @@ public TransformInfo(Guid id, RigidBodyTransform transform, MotionType mType)
-						{
-							indLast++;
-						}
+			public Guid SourceId;
 
-						// here merge the 2 lists items such that they are in an incremental order
-						bool r1 = indCurrent < snapshot.Transforms.Count;
-						bool r2 = indLast < lastsnapshot.Transforms.Count;
-						if (r1 && r2)
-						{
-							int cmpValue = snapshot.Transforms[indCurrent].Id.CompareTo(lastsnapshot.Transforms[indLast].Id);
-							if (cmpValue <= 0)
-							{
-								isEverythingSleeping = isEverythingSleeping && (snapshot.Transforms[indCurrent].motionType == MotionType.Sleeping);
-								mergedWithSleepingListTransforms.Add(snapshot.Transforms[indCurrent]);
-								indCurrent++;
-								indLast += (cmpValue == 0) ? 1 : 0;
-							}
-							else
-							{
-								isEverythingSleeping = isEverythingSleeping && (lastsnapshot.Transforms[indLast].motionType == MotionType.Sleeping);
-								mergedWithSleepingListTransforms.Add(lastsnapshot.Transforms[indLast]);
-								indLast++;
-							}
-						}
-						else
-						{
-							if (r1)
-							{
-								isEverythingSleeping = isEverythingSleeping && (snapshot.Transforms[indCurrent].motionType == MotionType.Sleeping);
-								mergedWithSleepingListTransforms.Add(snapshot.Transforms[indCurrent]);
-								indCurrent++;
-							}
-							else
-							{
-								if (r2)
-								{
-									isEverythingSleeping = isEverythingSleeping && (lastsnapshot.Transforms[indLast].motionType == MotionType.Sleeping);
-									mergedWithSleepingListTransforms.Add(lastsnapshot.Transforms[indLast]);
-									indLast++;
-								}
-							}
-						}
-					}
-#if MRE_PHYSICS_DEBUG
-					Debug.Log(" After merge: " + mergedWithSleepingListTransforms.Count + " before:" + snapshot.Transforms.Count
-						+ " last:" + lastsnapshot.Transforms.Count + " time:" + snapshot.Time + " isEverythingSleeping=" + isEverythingSleeping
-						+ " total size:" + Snapshots.Count);
-#endif
-					var snapshotExtended = new Snapshot(snapshot.Time, mergedWithSleepingListTransforms);
+			public RigidBodyTransform Transform;
 
-					// if all bodies were sleeping in the previous update then clear all the jitter buffer
-					if (areAllBodiesSleepingInTheLastSnapshot)
-					{
-#if MRE_PHYSICS_DEBUG
-						Debug.Log(" RESET JB  time:" + snapshot.Time + " isEverythingSleeping=" + isEverythingSleeping
-							+ " size:" + Snapshots.Count);
-#endif
-						Snapshots.Clear();
-						isBufferReseted = true;
-					}
-					
-					Snapshots.Add(snapshot.Time, snapshotExtended);
-					areAllBodiesSleepingInTheLastSnapshot = isEverythingSleeping;
-				}
-				else
-				{
-					Snapshots.Add(snapshot.Time, snapshot);
-					areAllBodiesSleepingInTheLastSnapshot = false;
-				}
-			}
-			return isBufferReseted;
+			public Vector3 LinearVelocity;
+
+			public Vector3 AngualrVelocity;
+
+			public MotionType MotionType;
+
+			public bool Updated;
 		}
 
+		public Guid Id { get; }
```

Comment:
This is also RigidBody ID ... ?

Reply:
No it's SourceID - other user ID

---
In reply to: [445059199](https://github.com/microsoft/mixed-reality-extension-unity/pull/199#discussion_r445059199) [](ancestors = 445059199)